resources/resource_utils.py

```diff
@@ -0,0 +1,60 @@
+from typing import Optional, Dict
+import hail as hl
+
+
+# Resource classes
+class _Resource:
```

Comment:
Should this be public? The base class could be useful for typing.

Reply:
yeah, why not? Should we use a different name then? AbstractResource or BaseResource ?

Reply:
I was thinking Resource without the underscore, since the underscore implies that it is not to be used outside this module.